erpnext/manufacturing/doctype/bom/bom.py

```diff
@@ -516,10 +516,16 @@ def company_currency(self):
 		return erpnext.get_company_currency(self.company)
 
 	def add_to_cur_exploded_items(self, args):
-		if self.cur_exploded_items.get(args.item_code):
-			self.cur_exploded_items[args.item_code]["stock_qty"] += args.stock_qty
+		if self.cur_exploded_items.get(args.item_code) and self.cur_exploded_items[args.item_code]["operation"] == args.operation:
+			if args.operation is None:
```

Comment:
at line number 519 you have checked self.cur_exploded_items[args.item_code]["operation"] == args.operation which means that  args.operation is not none then why you have added condition if args.operation is None at line number 520?

Reply:
In case the item does not have an operation attached to it,  both self.cur_exploded_items[args.item_code]["operation"] and args.operation will be none, so they will be equal. But in that case we dont want index to be ":item_code", just "item_code" hence the check for None.

Reply:
For that case you can check first args.operation is none or not at line 519, no need of extra condition in the next line 
like as below
```
self.cur_exploded_items.get(args.item_code) and args.operation and self.cur_exploded_items[args.item_code]["operation"] == args.operation:
```

Reply:
doing so overly complicated the logic
```
if self.cur_exploded_items.get(args.item_code) and args.operation and self.cur_exploded_items[args.item_code]["operation"] == args.operation:
	self.cur_exploded_items[args.item_code + ":" + args.operation]["stock_qty"] += args.stock_qty
elif self.cur_exploded_items.get(args.item_code) and not args.operation:
	self.cur_exploded_items[args.item_code]["stock_qty"] += args.stock_qty
elif args.operation:
	self.cur_exploded_items[args.item_code + ":" + args.operation] = args
else:
	self.cur_exploded_items[args.item_code] = args
```

Reply:
another solution is to use the :item_code for index when the operation is None. this will reduce the logic to simpler form.

Reply:
Try below one

```
key = (args.item_code)
if args.operation:
    key = (args.item_code, args.operation)

if key in self.cur_exploded_items:
    self.cur_exploded_items[key]["stock_qty"] += args.stock_qty
else:
    self.cur_exploded_items[key] = args
```